MatrixSDK/Space/MXSpaceService.swift

```diff
@@ -317,18 +367,54 @@ public class MXSpaceService: NSObject {
     
     // MARK: - Private
     
-    private func buildFlattenedParentIdList(with space: MXSpace, visitedSpaceIds: [String], flattenedParentIds: inout [String: Set<String>]) {
+    private func registerNotificationObservers() {
+        self.sessionStateDidChangeObserver = NotificationCenter.default.addObserver(forName: NSNotification.Name.mxSessionStateDidChange, object: session, queue: nil) { [weak self] notification in
+            guard let session = self?.session, session.state == .storeDataReady, let store = self?.session.store else {
+                return
+            }
+            
+            self?.loadData(from: store)
+        }
+        
+        self.sessionDidSyncObserver = NotificationCenter.default.addObserver(forName: NSNotification.Name.mxSessionDidSync, object: session, queue: nil) { [weak self] notification in
+            guard let self = self, let syncResponse = notification.userInfo?[kMXSessionNotificationSyncResponseKey] as? MXSyncResponse else {
```

Comment:
Note: if you need to call `buildGraph` in an accurate time during the processing of the sync response, you can make a call from `MXSession.handleSyncResponse()` like before. The execution order will be more controlled than via a notification.
But it seems that the space service has its own life cycle with no strong dependency with the MXSession life cycle. So, notification is fine here.